### PR TITLE
#12009: remove ttnn.conv and ttnn.optimized_conv

### PR DESCRIPTION
**Pull request by bbradelTT**
### Ticket
Link to Github Issue https://github.com/tenstorrent/tt-metal/issues/12009

### Problem description
- there are two old convs, conv() and optimized_conv(), that are no longer supported and need to be deleted

### What's changed
- delete these convs from ttnn and the corresponding pybind
- delete the uses of these ttnn convs, including from tests and models
- update the workflow scripts to remove usage of the deleted tests and models

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/10709363917
- [x] Blackhole Post commit (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/10708771080
- [x] Model regression CI testing passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/10708790511/job/29692358782 same failure as main: https://github.com/tenstorrent/tt-metal/actions/runs/10710779836
- [x] New/Existing tests provide coverage for changes

Device perf passes: https://github.com/tenstorrent/tt-metal/actions/runs/10708751402
TG Nightly passes: https://github.com/tenstorrent/tt-metal/actions/runs/10708765778
Nightly same failures as main: https://github.com/tenstorrent/tt-metal/actions/runs/10708804778/job/29692409662
T3K frequent same failures as main: https://github.com/tenstorrent/tt-metal/actions/runs/10707620457

### CODE REVIEW

File: models/demos/resnet/demo/images/ILSVRC2012_val_00040000_n01797886.JPEG
Comment (tt-aho) [inline models/demos/resnet/demo/images/ILSVRC2012_val_00040000_n01797886.JPEG:1]: These images are used by ttnn_resnet as well for the demo.
Should move this to these to the ttnn_resnet folder instead and update any paths referring to this folder accordingly

**Review by tt-aho — CHANGES_REQUESTED** [review]

File: models/demos/resnet/demo/images/ILSVRC2012_val_00040000_n01797886.JPEG
Comment (bbradelTT) [reply models/demos/resnet/demo/images/ILSVRC2012_val_00040000_n01797886.JPEG:1]: Thanks for pointing this out.

Moved the needed files over to ttnn_resnet and updated the ttnn_resnet files.

There are no more references to the `/resnet/demo/images` folder:
```
/proj_sw/user_dev/bbradel/tt-metal/models > grep '\/resnet\/demo\/images' -r * | wc
      0       0       0
```

The test passes:

```
pytest --disable-warnings models/demos/ttnn_resnet/demo/demo.py::test_demo_sample[16-models/demos/ttnn_resnet/demo/images/-device_params0]
...
====================================================== short test summary info =======================================================
PASSED models/demos/ttnn_resnet/demo/demo.py::test_demo_sample[16-models/demos/ttnn_resnet/demo/images/-device_params0]
=================================================== 1 passed, 2 warnings in 59.80s ===================================================
 ```

**Review by tt-aho — APPROVED** [review]

**Review by pavlejosipovic — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]
❤️

**Comment by sankarmanoj-tt** [comment]
There are a lot of files in `ttnn/cpp/ttnn/operations/conv/conv2d/device`  that are part of the old conv. Will you be deleting those in this PR?

**Review by mywoodstock — APPROVED** [review]
couple minor comments -- about adding the new ttnn_resnet in place of the old CI tests?

File: tests/scripts/run_performance.sh [hunk 1]
```diff
@@ -77,8 +77,6 @@ run_device_perf_models() {
         env pytest models/demos/ttnn_falcon7b/tests -m $test_marker --timeout=360
 
         env pytest models/demos/bert/tests -m $test_marker
-
-        env pytest models/demos/resnet/tests -m $test_marker
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply tests/scripts/run_performance.sh:81]: It's already on line 73.

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**